spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/orm/jpa/HibernatePropertiesTests.java

```diff
@@ -123,6 +123,15 @@ public void useNewIdGeneratorMappingsFalse() {
 										"false")));
 	}
 
+	@Test
+	public void useArchiveScanner() {
+		this.contextRunner.withPropertyValues(
+				"spring.jpa.hibernate.archive-scanner:org.hibernate.boot.archive.scan.internal.DisabledScanner")
+				.run(assertHibernateProperties((hibernateProperties) -> assertThat(
+						hibernateProperties).containsEntry(AvailableSettings.SCANNER,
```

Comment:
We need two tests here. One where we show this is set by default and one where we customize the property (for instance setting the standard implementation explicitly and then asserting we're using that.

Reply:
Added test for default implementation